addons/web/models/ir_http.py

```diff
@@ -17,6 +18,7 @@ class Http(models.AbstractModel):
 
     def webclient_rendering_context(self):
         return {
+            'get_modules_order': lambda: json_scriptsafe.dumps(module_boot()),
```

Comment:
This continues to feel awkward, like it should rather be a property (or lazyproperty) of the registry. And if it was, it could be accessed indirectly from any qweb template through `request.env.registry` without adding pollution to the rendering context. It could also be assigned when the registry has finished loading, instead of re-computing it every time it's accessed.

Let's say we make that `request.env.registry.modules`, a normal list of module names.
You'd use it in qweb templates like this:

```js
odoo._modules = <t t-raw="json.dumps(request.env.registry.modules)"/>;
```

(in this case `json` is `json_scriptsafe` already, as that's the only thing we expose to qweb)

What do you think?

Reply:
I have a checklist of all things that we need to do after this PR is merged (in https://pad.odoo.com/p/Assets_merge_checklist). I just added your comment on it. I promise that we will keep fixing issues/refactoring if necessary. We just need to merge this to stop blocking half the team rebasing and fixing conflicts left and right